src/comment.rs

```diff
@@ -811,7 +811,13 @@ impl<'a> CommentRewrite<'a> {
         } else if self.is_prev_line_multi_line && !line.is_empty() {
             self.result.push(' ')
         } else if is_last && line.is_empty() {
-            // trailing blank lines are unwanted
+            // Trailing blank lines are unwanted; if the last line is blank, look for additional,
+            // preceding blank lines to remove, also.
+            let trailing_line = self.comment_line_separator.trim_end_matches(' ');
```

Comment:
Would it be better to test for all whitespace characters?
```suggestion
            let trailing_line = self.comment_line_separator.trim_end_matches(char::is_whitespace);
```

Reply:
So, `self.comment_line_separator` is initialized as:
https://github.com/rust-lang/rustfmt/blob/ea02de27ff4119615c94107c6acb403b59593dae/src/comment.rs#L614

For `SingleBullet` block comments, this ends up being `"\n * "`, and the spaces before the asterisk can vary depending on the comment's indentation. A blank line in a comment would be a line with only newline + indentation + asterisk, so I remove the trailing space before comparing it to the end of `self.result`.

I suppose it would be more exhaustive to call `self.result.trim_end().ends_with(trailing_line)` because we'd then catch lines that are trailing blank with trailing whitespace. The logic would be more complex, though, because I'd have to check whether there was or was not trailing space when calculating the truncation length. I'll ruminate on the best way to do this, but let me know if you have any suggestions. Thanks!

To answer your question about how **all** extra blank lines are being removed, it's done in the `while` loop that repeats until `self.result` does not end with `trailing_line`.